src/core/openthread-core-config.h

```diff
@@ -77,6 +77,8 @@
 #include "config/time_sync.h"
 #include "config/tmf.h"
 
+#include "openthread-thread_v1_2-core-default-config.h"
```

Comment:
Not sure about this header file name and its location.

The OT configuration options (and their default value) are specified in `src/core/config/{module_name}.h`.
Suggest following same pattern for CSL, maybe adding all CSL related configs under a header named `src/core/config/csl.h`?